keras_cv/layers/preprocessing/random_flip.py

```diff
@@ -78,82 +78,49 @@ def __init__(
                 "RandomFlip layer {name} received an unknown mode="
                 "{arg}".format(name=self.name, arg=mode)
             )
-        self.auto_vectorize = True
         self.bounding_box_format = bounding_box_format
+        self.rate = 0.5
```

Comment:
Should we upgrade this to a constructor argument too while we're at it?

Reply:
@LukeWood 
Yeah, I intentionally replaced `0.5` with `self.rate` to increase the configurability. In pytorch and popular preprocessing libraries like Albumentations, `p` is used to adjust the probability of flipping.

- https://pytorch.org/vision/main/generated/torchvision.transforms.RandomHorizontalFlip.html
- https://albumentations.ai/docs/api_reference/augmentations/geometric/transforms/#albumentations.augmentations.geometric.transforms.Flip

I suggest adding an argument called `rate` with a default value of `0.5`. What do you think?

```python
class RandomFlip(VectorizedBaseImageAugmentationLayer):
    def __init__(
        self, mode=HORIZONTAL, rate=0.5, seed=None, bounding_box_format=None, **kwargs
    ):
```